aws/sqs_event_manager/sqs_event_manager/app.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2025 SUSE LLC.  All rights reserved.
+#
+# This file is part of suse-saas-tools
+#
+# suse-saas-tools is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# mash is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with mash.  If not, see <http://www.gnu.org/licenses/>
+#
+"""
+Lambda to handle SQS messages from from
+AWS Metering/Entitlement Marketplace API's
+"""
+import json
+import logging
+
+from sqs_event_manager.queue import delete_message
+from sqs_event_manager.message import AWSSNSMessage
+
+logger = logging.getLogger()
+
+
+def lambda_handler(event, context):
+    """
+    Expects event type matching the SNS topic
+
+    {
+        'records': [
+            {
+                'message_id': 'str',
+                'receipt_handle': 'str',
+                'body': {
+                    'Type' : 'str',
+                    'MessageId : 'str',
+                    'TopicArn' : 'str',
+                    'Message' : {
+                        'action': 'str',
+                        'customer-identifier': 'str',
+                        'product-code': 'str',
+                    },
+                    'Timestamp' : 'str',
+                    'SignatureVersion' : 'str',
+                    'Signature' : 'str',
+                    'SigningCertURL' : 'str',
+                    'UnsubscribeURL' : 'str'
+                },
+                'attributes': {
+                    'ApproximateReceiveCount': 'str',
+                    'SentTimestamp': 'str',
+                    'SenderId': 'str',
+                    'ApproximateFirstReceiveTimestamp': 'str'
+                },
+                'messageAttributes': {},
+                'md5OfBody': 'str',
+                'eventSource': 'aws:sqs',
+                'eventSourceARN': 'arn:aws:sqs:us-east-1:111122223333:my-queue',
+                'awsRegion': 'us-east-1'
+            }
+        ]
+    }
+    """
+    batch_item_failures = []
+    sqs_batch_response = {}
+
+    try:
+        for message in event['Records']:
```

Comment:
not sure and just to double check. Some AWS API's send responses base64 encoded. It was the case for the API Gateway. In case of SQS events can you make sure they will be never base64 encoded ?

Reply:
The example requests I was using are based on the previous iteration of code. And there is no base64 handling. We will need to do some testing against a test offer to be sure what the expected format is.

Reply:
ok :+1: